crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,353 @@
+                Some(attr) => {
+                    let role_value = attr.as_static_value()?;
+                    let role_text = role_value.text();
+
+                    if has_valid_role(role_text) || aria_roles.is_role_interactive(role_text) {
+                        return None;
+                    }
+                }
+                None => {
+                    return Some(());
+                }
+            };
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {{"Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element"}},
+        ))
+    }
+}
+
+/// Check if any interactive handler is present in the attributes.
+fn is_interactive_handler_present(attributes: &FxHashMap<String, Vec<String>>) -> bool {
+    let categories_to_check = vec!["focus", "keyboard", "mouse"];
+    for category in categories_to_check {
+        if let Some(handlers) = INTERACTIVE_HANDLER_LIST.get(category) {
+            for handler in handlers {
+                if let Some(values) = attributes.get(*handler) {
+                    if values.iter().any(|value| value != "null") {
+                        return true;
+                    }
+                }
+            }
+        }
+    }
+    false
+}
+
+/// This method is an override of AriaService::extract_attributes.
+///
+/// This modified version can extract attributes with dynamic values as well.
+/// The original extract_attributes in AriaService skips extracting other static attributes if a dynamic value is encountered.
+pub fn extract_attrs(attribute_list: &JsxAttributeList) -> Option<FxHashMap<String, Vec<String>>> {
+    let mut defined_attributes: FxHashMap<String, Vec<String>> = FxHashMap::default();
+    for attribute in attribute_list {
+        if let AnyJsxAttribute::JsxAttribute(attr) = attribute {
+            let name = attr.name().ok()?.syntax().text_trimmed().to_string();
+            let values = if let Some(initializer) = attr.initializer() {
+                let initializer = initializer.value().ok()?;
+                if let Some(static_value) = initializer.as_static_value() {
+                    static_value
+                        .text()
+                        .split(' ')
+                        .map(|s| s.to_string())
+                        .collect::<Vec<String>>()
+                } else {
+                    vec![initializer.syntax().text().to_string()]
+                }
+            } else {
+                vec!["true".to_string()]
+            };
+
+            defined_attributes.entry(name).or_insert(values);
+        }
+    }
+    Some(defined_attributes)
+}
```

Comment:
How about updating the original one?

Reply:
@unvalley 
Understood.
I will update the original method, and if there are any differences in the snapshot, I will handle it with a separate issue and PR.

Reply:
@unvalley 
fixed in this commit 3a83fabbaaf5b789c0d0eaf5a156d65f44cb9e69

No snapshot differences occurred.